cmake/TestSetup.cmake

```diff
@@ -47,17 +47,24 @@ endfunction()
 # Adds test that will run in serial
 # test output will be sent to /tst/regression/outputs/dir
 # test property labels: regression, mpi-no
-function(setup_test dir arg extra_labels)
+function(setup_test_serial dir arg extra_labels)
   separate_arguments(arg) 
   list(APPEND labels "regression;mpi-no")
   list(APPEND labels "${extra_labels}")
   if (Kokkos_ENABLE_OPENMP)
     set(PARTHENON_KOKKOS_TEST_ARGS "${PARTHENON_KOKKOS_TEST_ARGS} --kokkos-threads=${NUM_OMP_THREADS_PER_RANK}")
   endif()
-  add_test( NAME regression_test:${dir} COMMAND ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" 
-    ${arg} --test_dir "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}"
-    --output_dir "${PROJECT_BINARY_DIR}/tst/regression/outputs/${dir}"
-    --kokkos_args=${PARTHENON_KOKKOS_TEST_ARGS})
+  if (SERIAL_WITH_MPIEXEC)
+    set(PREFIX_SERIAL_REGRESSION_TEST ${TEST_MPIEXEC})
+    set(SUFFIX_SERIAL_REGRESSION_TEST --mpirun ${TEST_MPIEXEC})
+  endif()
+  add_test(
+    NAME regression_test:${dir}
+    COMMAND ${PREFIX_SERIAL_REGRESSION_TEST} ${Python3_EXECUTABLE} "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" 
```

Comment:
I'm not an expert in MPI contexts, but if I read this correct this means that we'd run python with h5py (that potentially and initializes mpi4py) to run another program (our examples) also with mpi.
Do I read this correctly? While this may work this somehow asks for trouble, doesn't it?
Also, if that's to only fix the `import h5py` issues, there may be other ways to address that (see https://github.com/h5py/h5py/issues/917 and related issues)

Reply:
Actually, it may not be necessary here.

Reply:
Let me check.

Reply:
*it being running python under mpiexec. Still need it for the test execution for sure.

Reply:
Removed the mpiexec call for running the python script - I didn't actually need that.

Reply:
Great!